lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -1,90 +1,69 @@
 package org.enso.syntax2;
 
-import java.io.File;
-import java.net.URISyntaxException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
-import java.nio.charset.StandardCharsets;
 
 public final class Parser implements AutoCloseable {
-  static {
-    String os = System.getProperty("os.name");
-    String name;
-    if (os.startsWith("Mac")) {
-      name = "libenso_parser.dylib";
-    } else if (os.startsWith("Windows")) {
-      name = "enso_parser.dll";
-    } else {
-      name = "libenso_parser.so";
-    }
-
-    File parser = null;
-    try {
-      var whereAmI = Parser.class.getProtectionDomain().getCodeSource().getLocation();
-      File dir = new File(whereAmI.toURI()).getParentFile();
-      parser = new File(dir, name);
-      System.load(parser.getAbsolutePath());
-    } catch (URISyntaxException | LinkageError e) {
-      File root = new File(".").getAbsoluteFile();
-      if (!searchFromDirToTop(e, root, "target", "rust", "debug", name)) {
-        throw new IllegalStateException("Cannot load parser from " + parser, e);
-      }
-    }
-  }
-
-  private static boolean searchFromDirToTop(Throwable chain, File root, String... names) {
-    while (root != null) {
-      var parser = root;
-      for (var e : names) {
-        parser = new File(parser, e);
-      }
-      try {
-        System.load(parser.getAbsolutePath());
-        return true;
-      } catch (LinkageError err) {
-        while (chain.getCause() != null) {
-          chain = chain.getCause();
-        }
-        chain.initCause(err);
-        root = root.getParentFile();
-      }
-    }
-    return false;
-  }
-
   private long state;
 
   private Parser(long stateIn) {
     state = stateIn;
   }
 
-  private static native long allocState();
-
-  private static native void freeState(long state);
+  private static long allocState() {
+    return 0;
+  }
 
-  private static native ByteBuffer parseInput(long state, ByteBuffer input);
+  private static void freeState(long state) {}
+
+  private static ByteBuffer parseInput(long state, ByteBuffer input) {
+    // fake output representing simple Enso source:
+    // main = 42
+    return ByteBuffer.wrap(
```

Comment:
@kazcw can you provide me a little bit of guidance? I need to know:
- how to compile WASM version of the parser?
- how to load it by JavaScript code?
- how to call `parseInput` (in JavaScript)?
- how to call `getLastInputBase` & other methods?

With that information I will be able to integrate the Enso Compiler with its `IR` with real parser in the browser. Thank you for hints & directions.

Reply:
Here's a JS port of the JNI API: https://github.com/enso-org/enso/tree/wip/kw/jni-js/lib/rust/parser/jni-js

It can be built like `cd lib/rust/parser/jni-js; wasm-pack build --target web`. For the most part functions correspond 1:1 with the JNI interface, however the way objects are allocated is different: `parse_input` returns a `ParseResult` object, which is passed to functions such as `base()` (equivalent of `getLastInputBase`) and `ast()` (the binary data). See the generated `.d.ts` file for the full API. Let me know if you have any questions.

Reply:
Am I doing something wrong? 

![null pointer passed to rust](https://github.com/enso-org/enso/assets/26887752/89810b17-ba49-483f-ac43-e49fe61c5503)

I can invoke `parse_input`, but I cannot pass the returned value to `ast`.